atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -122,6 +130,13 @@ public synchronized void updateCacheWithCommitTimestampsInformation(Set<Long> st
         startTimestamps.forEach(this::processCommitUpdate);
     }
 
+    @Override
+    public void requestStateRemoved(long startTs) {
+        // Ignoring the returned Future here is intentional, as the guarantees of this method are only that the
+        // associated state is eventually cleaned up (not necessarily immediately).
+        cleanupBatcher.apply(startTs);
```

Comment:
what happens if the async thing throws? previously the synchronous call to `ensureStateRemoved` would have also thrown and propagated?

Reply:
Good question. In general it shouldn't throw, as the cleanup here just does a bunch of in-memory data structure removals.

You're right to be cautious: that if it did (e.g. OOM, or a bug in our cleanup code), there is actually a semantic change here - we'd previously have failed the user request while now we'd pass it (we have to, since we no longer wait for this). I think this is fine, though.

Reply:
Ack, just wanted to make sure this didn't have correctness concerns.